src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +119,58 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer(this._pollingInterval);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string, timer?: number) {
+        if (!timer) {
+            timer = 0;
+        }
+        this._pollingInterval = window.setTimeout(async () =>  {
+            try {
+                if (timer !== undefined) {
+                    timer += POLLING_INTERVAL;
+                }
+                if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
```

Comment:
I think it is worthy to move this `if` before `window.setTimeout` ( as this logic should not be started if use `PAYPAL_COMMERCE_ALTERNATIVE_METHODS`) or even in the places where it could be triggered
like `onClick`

Reply:
fixed